### ISSUE TITLE
Answer YouTube question from Stevan B

### ISSUE THREAD

**Issue by dfarrell07** [submariner-io/submariner-website#208]
We've been asked the following question on one of our YouTube videos and likely should put together a reply.

```
The project looks cool, but the explanation about how it works lacks information to kind of put the pieces together.

Some of my questions, as a network engineer are:
- How many IPs need to be reachable from the Clusters to establish the IPSec tunnel? Which IP address is it? Is it a service IP or a Cluster IP? Is the broker using PODs behind the scenes?
- It was explained that only two Clusters could be used, one in each "Data Center" to establish the peering. Could I define how many NODEs of the existent cluster would act as the broker?
- Is there the possibility to specify the parameters used for the IPSec tunnel establishment such as encryption, authentication, etc?

Basically, the way I understood, based on the explanation is that the Broker communicates with the cluster, learns the IP addresses in use and is able to share the learned prefixes (POD and Cluster IP prefixes) through an IPSec connection to the other Cluster/Broker, so that both sides/clusters would be able to reach one another.

I hope I understood things correctly.
```

https://www.youtube.com/watch?v=jMvuchSMCKU&lc=UgzYPvtkiRlMJL7bIe94AaABAg

**Comment by sridhargaddam** [comment]
> Some of my questions, as a network engineer are:
> - How many IPs need to be reachable from the Clusters to establish the IPSec tunnel? 

For the IPsec tunnel to be established, atleast one end of the Cluster GatewayNode should have external reachability.
Assuming that you want to have a single GatewayNode in your clusters from which IPsec tunnels are setup, let me explain with some examples:

**Scenario1**: Say, you have two clusters, one is onPremise and the other one on Public cloud.
In this scenario, the GatewayNode on Public cloud should have a public-ip so that OnPrem cluster GatewayNode can connect to the public-ip. IPsec supports NAT-T, so onPrem cluster need **not** have public-ip.

**Scenario2**: If you have three clusters, one is onPrem and the other two are on Public cloud.
In this scenario, the Gateway nodes on each of the public cloud clusters need public-ip. 

Note: If you want to have multiple GatewayNodes in your public cluster (for Active/Passive HA), then each GatewayNode needs a public-ip.

> Which IP address is it? Is it a service IP or a Cluster IP? Is the broker using PODs behind the scenes?

No, Submariner does not install any Pods on the Broker cluster.
For more details, please refer to the following architecture details: https://submariner.io/architecture/broker/

> - It was explained that only two Clusters could be used, one in each "Data Center" to establish the peering. 

I'm afraid this is not the case. Submariner indeed supports connecting more than 2 clusters. It creates a full-mesh of IPsec tunnels between all the connected clusters.

> Could I define how many NODEs of the existent cluster would act as the broker?

I'm not sure I got your question. Broker is just a datastore to facilitate exchange of metadata information between participating clusters. The Broker cluster should be accessible from all the member clusters.
Please take a look at the architecture document for more details - https://submariner.io/architecture/

> - Is there the possibility to specify the parameters used for the IPSec tunnel establishment such as encryption, authentication, etc?

Currently, we don't support user-defined parameters for IPsec tunnel configuration.

**Comment by dfarrell07** [comment]
Thanks @sridhargaddam! I shared your answer with Stevan using the official account.

https://www.youtube.com/watch?v=jMvuchSMCKU&lc=UgzYPvtkiRlMJL7bIe94AaABAg.9AVjMuNyu9X9CTtlHqxALp
